surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/TestStrategyApplier.java

```diff
@@ -48,15 +54,28 @@ private boolean isUsageSet() {
         for (final String strategy : strategies) {
 
             final TestExecutionPlanner plannerForStrategy = testExecutionPlannerLoader.getPlannerForStrategy(strategy);
-            final List<? extends Class<?>> tests = plannerForStrategy.getTests().stream().map(testClass -> {
-                try {
-                    return Class.forName(testClass);
-                } catch (ClassNotFoundException e) {
-                    throw new IllegalStateException("Failed while obtaining strategy for " + strategy, e);
-                }
-            }).collect(Collectors.toList());
+            final List<? extends Class<?>> tests = plannerForStrategy.getTests()
+                .stream().filter(this::validateClassPathAndTestClass)
+                .map(testClass -> {
+                    try {
+                        return Class.forName(testClass);
+                    } catch (ClassNotFoundException e) {
+                        throw new IllegalStateException("Failed while obtaining strategy for " + strategy, e);
+                    }
+                }).collect(Collectors.toList());
             orderedTests.addAll(tests);
         }
         return orderedTests;
     }
+
+    private boolean validateClassPathAndTestClass(String testClass) {
+        Class<?> aClass;
+        try {
+            aClass = testClassLoader.loadClass(testClass);
+        } catch (ClassNotFoundException e) {
+            return false;
+        }
+
+        return jUnit4TestChecker.accept(aClass);
```

Comment:
I think we should just base it on naming conventions (so includes/excludes). Does the classpath bit fixes module issue? If so maybe we can copy the code to our own class to be free from this dependency?